### DISCUSSION TITLE
How to send `cta_url` with Image with caption?

### DISCUSSION THREAD

**Discussion by AliAryanTech** [wapikit/wapi.js#43] [Q&A] [answered]
Well, literally I'd look around but could not find it yet...

Send a image message with some caption and last add 2 urls button, is other buttons possible with it?
we can do with third party but if official one!?

**Comment by sarthakjdev** [comment, marked as answer]
In official WhatApp cloud API, it can be done using a interactive message of type button. 

You may check the official documentation [here](https://developers.facebook.com/docs/whatsapp/cloud-api/reference/messages#interactive-object).

I can see we missed the header object in the [ButtonInteractiveMessage](https://github.com/wapikit/wapi.js/blob/1b066ed4efaf997a4bb012da51514dfcb4165da6/packages/wapi.js/src/structures/interaction/interface.ts#L25) 

creating an issue for this. Adding high priority to it. 

We will try to fix it as soon as possible. 
If in case you wanna pick it up, drop a comment, I will assign you the issue. 


If you face any issue in setting up development environment, ask me, I will help you out.

**Reply by AliAryanTech** [reply]
alright thanks make sure you add a example code in docs too.

**Reply by AliAryanTech** [reply]
@sarthakjdev can you provide some example code using reply button and cta_url button? 

tell the diff in `replybuttoninteraction` & `QuickReplyButtonInteraction`

**Reply by sarthakjdev** [reply]
`ReplyButtonInteraction` is when reply button is clicked from the users end and user types some message as a reply. 
`QuickReplyButtonInteraction ` is when a button of type quickReply (having defined reply payload sent pre-configured) has been clicked.

> can you provide some example code using reply button and cta_url button? 

I will add it here for you in a while.
